src/meshes/parallel_tree.jl

```diff
@@ -55,44 +55,52 @@ mutable struct ParallelTree{NDIMS} <: AbstractTree{NDIMS}
         t.child_ids = fill(typemin(Int), 2^NDIMS, capacity + 1)
         t.neighbor_ids = fill(typemin(Int), 2 * NDIMS, capacity + 1)
         t.levels = fill(typemin(Int), capacity + 1)
-        t.coordinates = fill(NaN, NDIMS, capacity + 1)
+        t.coordinates = fill(convert(RealT, NaN), NDIMS, capacity + 1) # `NaN` is of type Float64
         t.original_cell_ids = fill(typemin(Int), capacity + 1)
         t.mpi_ranks = fill(typemin(Int), capacity + 1)
 
         t.capacity = capacity
         t.length = 0
         t.dummy = capacity + 1
 
-        t.center_level_0 = SVector(ntuple(_ -> NaN, NDIMS))
-        t.length_level_0 = NaN
+        t.center_level_0 = SVector(ntuple(_ -> convert(RealT, NaN), NDIMS))
+        t.length_level_0 = convert(RealT, NaN)
 
         return t
     end
 end
 
-# Constructor for passing the dimension as an argument
-ParallelTree(::Val{NDIMS}, args...) where {NDIMS} = ParallelTree{NDIMS}(args...)
+# Constructor for passing the dimension as an argument. Default datatype: Float64
+ParallelTree(::Val{NDIMS}, args...) where {NDIMS} = ParallelTree{NDIMS, Float64}(args...)
 
 # Create and initialize tree
-function ParallelTree{NDIMS}(capacity::Int, center::AbstractArray{Float64},
-                             length::Real, periodicity = true) where {NDIMS}
+function ParallelTree{NDIMS, RealT}(capacity::Int, center::AbstractArray{RealT},
+                                    length::RealT,
+                                    periodicity = true) where {NDIMS, RealT <: Real}
     # Create instance
-    t = ParallelTree{NDIMS}(capacity)
+    t = ParallelTree{NDIMS, RealT}(capacity)
 
     # Initialize root cell
     init!(t, center, length, periodicity)
 
     return t
 end
 
+function ParallelTree{NDIMS}(capacity::Int, center::AbstractArray{RealT},
+                             length::RealT,
+                             periodicity = true) where {NDIMS, RealT <: Real}
+    ParallelTree{NDIMS, RealT}(capacity, center, length, periodicity)
+end
+
 # Constructor accepting a single number as center (as opposed to an array) for 1D
-function ParallelTree{1}(cap::Int, center::Real, len::Real, periodicity = true)
-    ParallelTree{1}(cap, [convert(Float64, center)], len, periodicity)
+function ParallelTree{1, RealT}(cap::Int, center::RealT, len::RealT,
+                                periodicity = true) where {RealT <: Real}
+    ParallelTree{1, RealT}(cap, [center], len, periodicity)
```

Comment:
Like serial tree it would be better to have an interface `ParallelTree{1}` - even though there is no precompile for `ParallelTree{1}`